bigquery_etl/query_scheduling/dag.py

```diff
@@ -0,0 +1,63 @@
+"""Represents an Airflow DAG."""
+
+
+class DagParseException(Exception):
+    """Raised when DAG config is invalid."""
+
+    def __init__(self, message):
+        """Throw DagParseException."""
+        message = f"""
+        {message}
+
+        Expected yaml format:
+        name:
+            schedule_interval: string,
+            default_args: map
+        """
+
+        super(DagParseException, self).__init__(message)
+
+
+class Dag:
+    """Representation of a DAG configuration."""
+
+    def __init__(self, name, schedule_interval, default_args):
+        """Instantiate new DAG representation."""
+        self.name = name
+        self.schedule_interval = schedule_interval
+        self.default_args = default_args
+        self.tasks = []
+
+    def add_tasks(self, tasks):
+        """Add tasks to be scheduled as part of the DAG."""
+        self.tasks += tasks
+
+    @classmethod
+    def from_dict(cls, d):
+        """
+        Parse the DAG configuration from a dict and create a new Dag instance.
+
+        Expected dict format:
+        {
+            "name": {
+                "schedule_interval": string,
+                "default_args": dict
+            }
+        }
+        """
+        if len(d.keys()) != 1:
+            raise DagParseException(f"Invalid DAG name in {d}.")
+
+        name = list(d.keys())[0]
+
+        if "schedule_interval" not in d[name]:
+            raise DagParseException(f"schedule_interval missing in {d}.")
```

Comment:
It's looking like we have a good bit of validation code here. We might want to consider defining JSON schemas for the expected structure and rely on validating against the schema rather than writing our own validation code. We'd write the json schemas in yaml files like is done for Glean. We can write up a separate issue for that if you think it's worth pursuing.

Reply:
This might be a good idea, especially as the config files get more complex. I'll create a separate issue for this.